Atividades/para-casa-roberta/exercicio.js

```diff
@@ -0,0 +1,165 @@
+class AnimalDomestico {
+  constructor(nome, idade, cor, castrado) {
+    this.nome = nome;
+    this.idade = idade;
+    this.cor = cor;
+    this.consultas = [];
+    this.vacinas = [];
+    this.castrado = castrado;
+  }
+}
+
+class Gato extends AnimalDomestico {
+  constructor(nome, idade, cor, castrado, externo, social) {
+    super(nome, idade, cor, castrado);
+    this.externo = externo;
+    this.social = social;
+    this.consultas = [];
+    this.vacinas = [];
+  }
+  miar() {
+    console.log("miau miau");
+  }
+  brincar() {
+    console.log(`${this.nome} use uma varinha`);
+  }
+  acariciar() {
+    if (this.social == true) {
+      console.log("ron ron ron");
+    } else {
+      console.log("schiiii");
+    }
+  }
+  vacinar(vacina) {
+    this.vacinas.push(vacina);
+    this.consultas.unshift(new Date());
+  }
+
+  castrar() {
+    if (this.castrado == true) {
+      console.log(` ${this.nome}, já está castrado.`);
+    } else {
+      console.log(`${this.nome}, foi castrado`);
+      this.consultas.unshift(new Date());
+      return (this.castrado = true);
+    }
+  }
+
+  alimentar() {
+    if (this.social == false) {
+      console.log(`${this.nome} recebe um delicioso peixe!`);
+      return this.social == true;
+    }
+  }
```

Comment:
Fiz de uma forma diferente! Coloquei, para quando comer frango, sachê ou peixe, o gato se tornar sociável. Vou te mostrar:

    alimentar(alimento) {
        if (alimento == "frango") {
            this.sociavel = true
        } else if (alimento == "sachê") {
            this.sociavel = true
        } else if (alimento == "peixe") {
            this.sociavel = true
        } else {
            return
        }
    }

Reply:
Obrigada pela sugestão!